libraries/common/components/InfiniteContainer/index.jsx

```diff
@@ -213,6 +224,11 @@ class InfiniteContainer extends Component {
   allItemsAreRendered(props = this.props) {
     const [start, length] = this.state.offset;
 
+    if (props.promiseBased) {
+      const { totalItems } = props;
+      return totalItems !== null && (start + length >= totalItems);
+    }
+
```

Comment:
I had to change the handle function a bit, so that offset increased after the request. so the part `start + length` was updated to just `start` or `offset`